src/Compilers/Core/Portable/Diagnostic/SuppressionDescriptor.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Provides a description about a programmatic suppression of a <see cref="Diagnostic"/> by a <see cref="DiagnosticSuppressor"/>.
+    /// </summary>
+    public sealed class SuppressionDescriptor : IEquatable<SuppressionDescriptor>
```

Comment:
Should this be a struct?

Reply:
I have kept this identical to [DiagnosticDescriptor](http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/DiagnosticDescriptor.cs,14). @gafter for further comment.

Reply:
Filed https://github.com/dotnet/roslyn/issues/36568 to track the suggestion and unblock the PR.